app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/FahrplanFragment.java

```diff
@@ -298,21 +317,39 @@ private void updateNavigationMenuSelection() {
         }
     }
 
-    private void addRoomColumns(HorizontalSnapScrollView scroller) {
-        LinearLayout root = (LinearLayout) scroller.getChildAt(0);
-        root.removeAllViews();
+    private void addRoomColumns(@NonNull HorizontalSnapScrollView scroller, @NonNull List<Lecture> lectures, int roomCount) {
         if (scroller.getColumnWidth() != 0) {
             // update pre-calculated width with actual layout
             columnWidth = scroller.getColumnWidth();
         }
-        for (int i = 0; i < MyApp.roomCount; i++) {
-            LinearLayout roomLayout = new LinearLayout(context);
-            LinearLayout.LayoutParams p = new LayoutParams(
-                    columnWidth, LayoutParams.MATCH_PARENT, 1);
-            roomLayout.setOrientation(LinearLayout.VERTICAL);
-            roomLayout.setLayoutParams(p);
-            root.addView(roomLayout);
+
+        LinearLayout columnsLayout = (LinearLayout) scroller.getChildAt(0);
+        columnsLayout.removeAllViews();
+
+        LecturesPerRoom lecturesPerRoom = new LecturesPerRoom(lectures);
+        int boxHeight = getNormalizedBoxHeight(getResources(), scale, LOG_TAG);
+        LayoutCalculator layoutCalculator = new LayoutCalculator(Logging.Companion.get(), boxHeight);
+        for (int roomIndex = 0; roomIndex < roomCount; roomIndex++) {
+            boolean roomHasLecture = lecturesPerRoom.containsKey(roomIndex);
+            if (!roomHasLecture) {
+                continue;
+            }
+            List<? extends Lecture> roomLectures = lecturesPerRoom.get(roomIndex);
```

Comment:
:mag: You can avoid the `List<? extends Lecture>` type declaration by adding this getter to the `LecturesPerRoom` class:

``` kotlin
override fun get(key: Int) = super.get(key)
```